grants_tagger/train.py

```diff
@@ -223,14 +225,44 @@ def train_and_evaluate(
             label_binarizer=label_binarizer,
             from_same_distribution=from_same_distribution
         )
-        model.fit(X_train, Y_train)
+        if y_batch_size:
+            vectorizer = model.steps[0][1]
+            classifier = model.steps[1][1]
+
+            Path(model_path).mkdir(exist_ok=True)
+            print("Fitting vectorizer")
+            vectorizer.fit(X_train)
+            with open(f"{model_path}/vectorizer.pkl", "wb") as f:
+                f.write(pickle.dumps(vectorizer))
+            print("Training model")
+            for tag_i in range(0, Y_train.shape[1], y_batch_size):
+                print(tag_i)
+                X_train_vec = vectorizer.transform(X_train)
+                classifier.fit(X_train_vec, Y_train[:,tag_i:tag_i+512]) # assuming that fit clears previous params
```

Comment:
should this be `Y_train[:,tag_i:tag_i+y_batch_size]` ? I am not sure if I understand the 512.

Reply:
i will replace it with `y_batch_size`, is that clearer?

Reply:
Resolved by 2701cf6ab0b2232256378fe5a327d940f242c36e